examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+   "metadata": {},
+   "source": [
+    "If you don't want to re-run the python code after making changes to the file that contains the component, you can see changes happening immediately thanks to **hot-reloading**.\n",
+    "This requires `pip install watchdog`.  \n",
+    "Next, you replace the line  \n",
+    "`_esm = pathlib.Path(\"my_component.tsx\").read_text()`  \n",
+    "with   \n",
+    "`_esm = pathlib.Path(\"my_component.tsx\")`  \n",
+    "\n",
+    "Now open `my_component.tsx`, change \"Hello World\" to \"Hi there\", and you will see that the changes are reflected immediately."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d7bd0c3c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "import pathlib\n",
+    "\n",
+    "class WidgetFromFile(ipyreact.ReactWidget):\n",
+    "    _esm = pathlib.Path(\"my_component.tsx\") # <- this will not work in JupyterLite\n",
+    "\n",
+    "WidgetFromFile()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "82c9fbe3-6112-4dae-b817-e1a12343b407",
+   "metadata": {},
+   "source": [
+    "one more thing:  \n",
+    "Having **autocompletion in IDEs** is awesome!  \n",
+    "traitlets don't have that by default, but adding a `signature_has_traits` decorator will do the job!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "89793f7c-1398-4f11-ab75-b73ffacd9b43",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "from traitlets import Any, Unicode, Int, observe, signature_has_traits\n",
+    "\n",
+    "@signature_has_traits\n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    my_width = Int(23).tag(sync=True)\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton({ my_width }) {\n",
+    "      return (\n",
+    "        <button\n",
+    "          style={{\n",
+    "            position: \"relative\",\n",
+    "            width: my_width,\n",
+    "            height: 30,\n",
+    "          }}\n",
+    "        >\n",
+    "          {\" \"}\n",
+    "          Width of {my_width} px{\" \"}\n",
+    "        </button>\n",
+    "      );\n",
+    "    }\"\"\"\n",
+    "MyExampleWidget(my_width=300)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "38cfce36",
+   "metadata": {},
+   "source": [
+    "And this screenshots shows that autocompletion works now:  \n",
+    "<img src=\"autocomplete_screenshot.png\" width=\"400\">"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "bae77944-e0dd-49c4-bd27-b401d7bbef1b",
+   "metadata": {},
+   "source": [
+    "Now we want to **print a message at class initialization** that says \"Button was initialized with width 300px!\"  \n",
+    "That is possible, but not very straight forward, so feel free to skip this example."
```

Comment:
What part isn't straightforward, the `super` call?    maybe explicitly mention that, along with `super` being required.